tests/mir-opt/matches_reduce_branches.rs

```diff
@@ -225,6 +225,30 @@ fn match_i8_i16_failed(i: EnumAi8) -> i16 {
     }
 }
 
+// We cannot transform it, even though `-1i8` and `255i16` are the same in terms of bits,
+// what we actually require is that they are considered equal in a signed comparison.
```

Comment:
```suggestion
// what we actually require is that they are considered equal in a signed comparison,
// after sign-extending to the larger type.
```
At least I think that's what happens?